audio-device-client/idl-example.md

```diff
@@ -0,0 +1,113 @@
+**Note: that this content is work-in-progress at the early stage of
+development.**
+
+
+# WebIDL (WIP)
+
+### AudioDeviceClientState
+An `enum` represents the current state of the client.
+
+```webidl
+enum AudioDeviceClientState {
+  "pending",
+  "running",
+  "stopped"
+}
+```
+
+### AudioDeviceClient
+
+The low-level representation of an audio device. Provides various
+hardware-related properties and a path to the associated `AudioContext` for Web
+Audio API tasks.
+
+```webidl
+[Exposed=Window, SecureContext, NoConstructor]
+interface AudioDeviceClient : EventTarget {
+  readonly attribute USVString deviceId;
+  readonly attribute float sampleRate;
+  readonly attribute unsigned callbackBufferSize;
+  readonly attribute unsigned inputChannelCount;
+  readonly attribute unsigned outputChannelCount;
+  readonly attribute AudioDeviceClientState state;
+  readonly attribute MessagePort port;
+  attribute EventHandler onstatechange;
+  Promise<void> start(USVString moduleUrl);
+  Promise<void> stop();
+  [SameObject] AudioContext getContext();
+}
+```
+
+### AudioDeviceClientConstraints
+
+A dictionary describes user-specified device constraints. It can be passed when
+`AudioDeviceClient` object is instantiated.
+
+```webidl
+dictionary AudioDeviceClientConstraints {
+  USVString deviceId;
+  optional float sampleRate;
+  unsigned callbackBufferSize = 128;
```

Comment:
How will this work on Android?  I don't see too many Android devices that actually have 128 as the buffer size.   I think this should be optional?

Reply:
Then perhaps it should be optional.